src/net_processing.cpp

```diff
@@ -848,13 +848,12 @@ void PeerManagerImpl::MaybeSetPeerAsAnnouncingHeaderAndIDs(NodeId nodeid)
             }
         }
         m_connman.ForNode(nodeid, [this](CNode* pfrom){
-            AssertLockHeld(cs_main);
+            LockAssertion lock(::cs_main);
```

Comment:
Important portions of the PR were omitted due to reliance of descriptor wallet definitions (3ddc150857178bfb1c854c05bf9b526777876f56), changes that would break building due to a need for Dash-specific changes (2ee7743fe723227f2ea1b031eddb14fc6863f4c8), therefore, only the most minimum required set of changes have been backported.